File: solutions/javascript/lasagna-master/1/lasagna-master.js
export function cookingStatus(minutes) {
  if (minutes === 0) {
    return 'Lasagna is done.';
  }

  if (typeof minutes === 'number') {
    return 'Not done, please wait.';
  }

  return 'You forgot to set the timer.';
}

export function preparationTime(layers, minutes = 2) {
  return layers.length * minutes;
}

export function quantities(layers) {
  let noodles = 0;
  let sauce   = 0;

  layers.forEach(layer => {
    if (layer === 'noodles') {
      noodles += 1;
    }

    if (layer === 'sauce') {
      sauce += 1;
    }
  });

  return {
    noodles: noodles * 50,
    sauce: sauce * 0.2
  };
}

export function addSecretIngredient(friendIngredients, myIngredients) {
  myIngredients.push(friendIngredients[friendIngredients.length -1]);
}

export function scaleRecipe(recipe, portionCount = 0) {
  let scaled = {};

  Object.keys(recipe).forEach(key => {
    scaled[key] = recipe[key] / 2.0 * portionCount;
  });

  return scaled;
}

File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

func PreparationTime(layers []string, minutesPerLayer int) int {
    if minutesPerLayer == 0 {
        minutesPerLayer = 2
    }

    return len(layers) * minutesPerLayer
}

func Quantities(layers []string) (noodles int, sauce float64) {
    noodleLayerCount := 0
    sauceLayerCount  := 0.0

    for _, layer := range layers {
        if layer == "noodles" {
            noodleLayerCount++
        }

        if layer == "sauce" {
            sauceLayerCount++
        }
    }

    noodles, sauce = noodleLayerCount * 50, sauceLayerCount * 0.2
    return
}

func AddSecretIngredient(friendIngredients []string, myIngredients []string) {
    friendLastIndex := len(friendIngredients) - 1
    myLastIndex     := len(myIngredients)     - 1

    myIngredients[myLastIndex] = friendIngredients[friendLastIndex]
}

func ScaleRecipe(quantities []float64, scale int) []float64 {
    scaledQuantities := make([]float64, len(quantities))
    copy(scaledQuantities, quantities)

    for i, quantity := range scaledQuantities {
        scaledQuantities[i] = quantity * float64(scale) / 2.0
    }

    return scaledQuantities
}
